django/bossingest/utils.py

```diff
@@ -0,0 +1,183 @@
+from bosscore.error import BossError, ErrorCodes, BossResourceNotFoundError
+import boto3
+import json
+from ingestclient.core.backend import BossBackend
+from ndingest.nddynamo.boss_tileindexdb import BossTileIndexDB, MAX_TASK_ID_SUFFIX, TASK_INDEX
+from ndingest.ndingestproj.bossingestproj import BossIngestProj
+from ndingest.ndqueue.uploadqueue import UploadQueue
+import os
+from tempfile import NamedTemporaryFile
+
+# First value should be job id.
+CSV_FORMAT_STR = 'tiles-{}&'
+
+# SQS Hardlimit
+SQS_BATCH_SIZE = 10
+SQS_RETRY_TIMEOUT = 15
+
+def query_tile_index(job_id, tile_index_name, region):
+    """
+    Find all chunks belonging to a job and output those that have more than
+    10 tiles to a .csv file named by the job_id.
+
+    Args:
+        job_id (int): Id of ingest job.
+        tile_index_name (str): Name of DynamoDB tile index table.
+
+    Returns:
+        (str|None): Filename of list of chunks that meet criteria or None.
+    """
+    client = boto3.client('dynamodb', region_name=region)
+
+    paginator = client.get_paginator('query')
+    work_to_do = False
+
+    with NamedTemporaryFile(
+        mode='wt',
+        prefix=CSV_FORMAT_STR.format(job_id),
+        suffix='.csv',
+        delete=False
+    ) as output_csv:
+        output_file = output_csv.name
+        for i in range(MAX_TASK_ID_SUFFIX):
+            response_iterator = _query_tile_index(paginator, tile_index_name, job_id, i)
+
+            for response in response_iterator:
+                for item in response["Items"]:
+                    if len(item["tile_uploaded_map"]["M"]) > 10:
```

Comment:
I think we may want to always mark a chunk for upload if it's still in the tile index now that we've moved deletion of the chunk before we remove the message from the queue.

Reply:
Agreed.

Reply:
Looking back at this, this change was not made before merging in.